pkg/compact/compact.go

```diff
@@ -470,6 +481,98 @@ func (cg *Group) Resolution() int64 {
 	return cg.resolution
 }
 
+// metrics related to planning and compaction progress
+type ProgressMetrics struct {
+	NumberOfIterations    *prometheus.GaugeVec
+	NumberOfBlocksToMerge *prometheus.GaugeVec
+	// figure out where to add groupKey labels to this
+}
+
+// should return the results/metrics of the planning simulation
+type PlanSim interface {
+	ProgressCalculate(ctx context.Context, groups []*Group) error
+}
+
+type DefaultPlanSim struct {
+	planner Planner
+	ProgressMetrics
+}
+
+func NewDefaultPlanSim(reg prometheus.Registerer, logger log.Logger) *DefaultPlanSim {
+	return &DefaultPlanSim{
+		planner: NewTSDBBasedPlanner(logger, []int64{}),
+		ProgressMetrics: ProgressMetrics{
+			NumberOfIterations: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+				Name: "thanos_number_of_iterations",
+				Help: "number of iterations to be done",
+			}, []string{"group"}),
+			NumberOfBlocksToMerge: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+				Name: "thanos_number_of_blocks_planned",
+				Help: "number of blocks planned to be merged",
+			}, []string{"group"}),
+		},
+	}
+}
+
+func (ps *DefaultPlanSim) ProgressCalculate(ctx context.Context, groups []*Group) error {
+	iterations := 0
+	blocksToMerge := 0
+	var groupCompactions, groupBlocks map[string]int
```

Comment:
Initialize maps explicitly with size

Reply:
I'll remember this from now on, I think this is part of the Style Guide too.